### PR TITLE
Feature/perf improvements

### PR DESCRIPTION
**Pull request by makosblade**
Lots of changes/additions here:
- added a generic valuemapper that uses a provided streamserializer to map values
- added rethinkmapstorefactorybuilder( a bit overengineered )
- added self registering valuemappers
- updated hazelcast and jetty
- made a new rhizomeutils and merged the old one into it
- added immutables(http://immutables.github.io/) library but not currently in use
- a bunch of minute stylistic cleanup (unused exceptions, unnecessary nesting, etc)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -66,12 +70,24 @@ repositories {
 ext.spring_security_version='4.0.2.RELEASE'
 ext.spring_framework_version='4.2.1.RELEASE'
 ext.jackson_version='2.4.4'
-ext.jetty_version='9.3.3.v20150827'
-ext.hazelcast_version='3.5.3'
+ext.jetty_version='9.3.6.v20151106'
```
Comment (UnsungHero97) [inline build.gradle]: so new! :)

**Comment by geekbeast** [comment]
LGTM. @makosblade  Okay to merge?

**Comment by makosblade** [comment]
I think so but lemme double check. I'll ping you in a few minutes if it's g2g

**Comment by makosblade** [comment]
Yep, should be g2g

**Pull request merged**